e2e/pod_status.go

```diff
@@ -122,110 +31,8 @@ func isIPPoolAllocationsEmpty(k8sIPAM *kubeClient.KubernetesIPAM, ipPoolName str
-		if err != nil && !k8serrors.IsNotFound(err) {
-			return false, fmt.Errorf("something weird happened with the stateful set whose status is: [%s]. Errors: %w", statefulSet.Status.String(), err)
-		}
-
-		associatedPods, err := cs.CoreV1().Pods(namespace).List(context.TODO(), selectViaLabels(labelSelector))
-		if err != nil {
-			return false, err
-		}
-
-		return isStatefulSetEmpty(statefulSet) && areAssociatedPodsGone(associatedPods), nil
-	}
-}
-
-func selectViaLabels(labelSelector string) metav1.ListOptions {
-	return metav1.ListOptions{LabelSelector: labelSelector}
-}
-
-func isStatefulSetEmpty(statefulSet *appsv1.StatefulSet) bool {
-	return statefulSet.Status.CurrentReplicas == int32(0)
-}
-
-func areAssociatedPodsGone(pods *v1.PodList) bool {
-	return len(pods.Items) == 0
-}
-
-// WaitForPodReady polls up to timeout seconds for pod to enter steady state (running or succeeded state).
-// Returns an error if the pod never enters a steady state.
-func WaitForPodReady(cs *kubernetes.Clientset, namespace, podName string, timeout time.Duration) error {
-	return wait.PollImmediate(time.Second, timeout, isPodRunning(cs, podName, namespace))
-}
-
-// WaitForPodToDisappear polls up to timeout seconds for pod to be gone from the Kubernetes cluster.
-// Returns an error if the pod is never deleted, or if GETing it returns an error other than `NotFound`.
-func WaitForPodToDisappear(cs *kubernetes.Clientset, namespace, podName string, timeout time.Duration) error {
-	return wait.PollImmediate(time.Second, timeout, isPodGone(cs, podName, namespace))
-}
-
-// This function only plays nice with the replicaSet it's being used with.
-// Any pods that might be up still from a previous test may cause unexpected results.
-func WaitForReplicaSetSteadyState(cs *kubernetes.Clientset, namespace, label string, replicaSet *appsv1.ReplicaSet, timeout time.Duration) error {
-	return wait.PollImmediate(time.Second, timeout, isReplicaSetSteady(cs, replicaSet.Name, namespace, label))
-}
-
-// WaitForReplicaSetToDisappear polls up to timeout seconds for replicaset to be gone from the Kubernetes cluster.
-// Returns an error if the replicaset is never deleted, or if GETing it returns an error other than `NotFound`.
-func WaitForReplicaSetToDisappear(cs *kubernetes.Clientset, namespace, rsName string, timeout time.Duration) error {
-	return wait.PollImmediate(time.Second, timeout, isReplicaSetGone(cs, rsName, namespace))
-}
-
 // WaitForZeroIPPoolAllocations polls up to timeout seconds for IP pool allocations to be gone from the Kubernetes cluster.
 // Returns an error if any IP pool allocations remain after time limit, or if GETing IP pools causes an error.
 func WaitForZeroIPPoolAllocations(k8sIPAM *kubeClient.KubernetesIPAM, ipPoolName string, timeout time.Duration) error {
 	return wait.PollImmediate(time.Second, timeout, isIPPoolAllocationsEmpty(k8sIPAM, ipPoolName))
 }
```

Comment:
I love the way you moved the code around for this refactor Miguel. However `pod_status.go` currently has no code related to pods. I propose we rename this file to `ippool_status.go` or something like that.

Reply:
Done.